src/FactSystem/ParameterLoader.cc

```diff
@@ -601,40 +602,42 @@ void ParameterLoader::_writeParameterRaw(int componentId, const QString& paramNa
     _vehicle->sendMessageOnLink(_vehicle->priorityLink(), msg);
 }
 
-void ParameterLoader::_writeLocalParamCache()
+void ParameterLoader::_writeLocalParamCache(int uasId, int componentId)
 {
-    QMap<int, MapID2NamedParam> cache_map;
+    MapID2NamedParam cache_map;
 
-    foreach(int component, _mapParameterId2Name.keys()) {
-        foreach(int id, _mapParameterId2Name[component].keys()) {
-            const QString name(_mapParameterId2Name[component][id]);
-            const Fact *fact = _mapParameterName2Variant[component][name].value<Fact*>();
-            cache_map[component][id] = NamedParam(name, ParamTypeVal(fact->type(), fact->rawValue()));
-        }
+    foreach(int id, _mapParameterId2Name[componentId].keys()) {
+        const QString name(_mapParameterId2Name[componentId][id]);
+        const Fact *fact = _mapParameterName2Variant[componentId][name].value<Fact*>();
+        cache_map[id] = NamedParam(name, ParamTypeVal(fact->type(), fact->rawValue()));
     }
 
-    QFile cache_file(QFileInfo(QSettings().fileName()).path() + QDir::separator() + "param_cache");
+    QFile cache_file(parameterCacheFile(uasId, componentId));
     cache_file.open(QIODevice::WriteOnly | QIODevice::Truncate);
 
     QDataStream ds(&cache_file);
     ds << cache_map;
 }
 
-QString ParameterLoader::parameterCacheFile(void)
+QDir ParameterLoader::parameterCacheDir()
+{
+    const QString spath(QFileInfo(QSettings().fileName()).dir().absolutePath());
+    return spath + QDir::separator() + "ParamCache";
+}
+
+QString ParameterLoader::parameterCacheFile(int uasId, int componentId)
 {
-    const QDir settingsDir(QFileInfo(QSettings().fileName()).dir());
-    return settingsDir.filePath("param_cache");
+    return parameterCacheDir().filePath(QString("%1_%2").arg(uasId).arg(componentId));
```

Comment:
Does this just leave the file name as #_#? Can't quite tell fron just the diff. If so, would still be good to have a user readable file name to identify file plus the vehicle id/compid.